### ISSUE TITLE
False Positive: `SlevomatCodingStandard.Classes.UnusedPrivateElements`  fails to detect class visibility

### ISSUE THREAD

**Issue by bcremer** [slevomat/coding-standard#625]
Given I have a Immutable Collection Class that return new instances of itself.
When I call private methods with the new objects
Then it should not report unused private methods.

Minimal testcase to reproduce:

```php
class Test
{

    public function testOnFilteredCollection() : self
    {
        $filtered = $this->filter();
        
        // $filtered is instance of Test so it can access private methods from class Test
        $filtered->testFoo();
    }

    private function filter() : self
    {
        // filter content, return new instance of itself
        return new self;
    }

    private function testFoo() : bool
    {
    }

}
```

**Expected:**
No errors or warnings.

**Actual:**
Error: `Class Test contains unused private method testFoo()`

I can provide a failing testcase for `\SlevomatCodingStandard\Sniffs\Classes\UnusedPrivateElementsSniffTest` if this is confirmed as a bug.

**Comment by Majkl578** [comment]
This goes beyond capabilities of PHPCS, you will need to exclude the rule for this file.

**Comment by bcremer** [comment]
Fair enough. Should I keep this issue open for future reference?
